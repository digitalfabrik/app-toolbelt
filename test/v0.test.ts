import buildCommand from '../src/core.js'
import { CommanderError } from 'commander'

const getCommandOutput = (args: readonly string[]) => {
  let output = ''
  let command = buildCommand(err => {
    throw err
  }).configureOutput({
    writeOut: (str: string) => {
      output += str
    },
    writeErr: (str: string) => {
      output += str
    },
    outputError: (str: string, write: (str: string) => void) => {
      output += str
    }
  })

  let argv = ['node', 'whitelabel', ...args]
  try {
    command.parse(argv)
  } catch (e) {
    if (!(e instanceof CommanderError)) {
      throw e
    }
  }

  return output.replace(/\s+/g, ' ').trim()
}

describe('stability test', () => {
  test('v0', () => {
    expect(getCommandOutput(['v0'])).toMatchSnapshot()
  })

  test('release', () => {
    expect(getCommandOutput(['v0', 'release'])).toMatchSnapshot()
  })

  test('release-notes', () => {
    expect(getCommandOutput(['v0', 'release-notes'])).toMatchSnapshot()
  })

  test('build-config', () => {
    expect(getCommandOutput(['v0', 'build-config'])).toMatchSnapshot()
  })

  test('version', () => {
    expect(getCommandOutput(['v0', 'version'])).toMatchSnapshot()
  })

  test('notify', () => {
    expect(getCommandOutput(['v0', 'notify'])).toMatchSnapshot()
  })
})
