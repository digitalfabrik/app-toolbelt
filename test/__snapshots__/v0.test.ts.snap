// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stability test build-config 1`] = `
"Usage: whitelabel v0 build-config [options] [command]

Options:
  -h, --help                                               display help for command

Commands:
  to-bash [options] <build-config-name> <platform>         outputs the specified build config as key-value pairs which can be executed by bash
  to-json [options] <build-config-name> <platform>         outputs the specified build config as JSON
  to-properties [options] <build-config-name> <platform>   create and write a new properties file to the stdout
  write-xcconfig [options] <build_config_name> <platform>  create and write a new buildConfig.tmp.xcconfig to the output directory
  help [command]                                           display help for command
"
`;

exports[`stability test ci 1`] = `
"Usage: whitelabel v0 ci [options] [command]

Options:
  -h, --help                         display help for command

Commands:
  trigger [options] <workflow-type>  trigger a workflow in the ci on the main
                                     branch
  help [command]                     display help for command
"
`;

exports[`stability test notify 1`] = `
"Usage: whitelabel v0 notify [options] [command]

Options:
  -h, --help            display help for command

Commands:
  mattermost [options]  Send a message to a mattermost channel
  help [command]        display help for command
"
`;

exports[`stability test release 1`] = `
"Usage: whitelabel v0 release [options] [command]

Options:
  -h, --help                                                                   display help for command

Commands:
  bump-to [options] <new-version-name> <new-version-code>                      commits the supplied version name and code to github and tags the commit
  create [options] <platform> <new-version-name> <new-version-code>            creates a new release for the specified platform
  upload [options]                                                             Upload a release asset to github
  promote [options]                                                            Remove pre-release flag from the latest release
  sentry-create [options] <package-name> <version-name> <sourcemap-directory>  create a new release on sentry. This file uses the configuration from ./.sentryclirc
  help [command]                                                               display help for command
"
`;

exports[`stability test release-notes 1`] = `
"Usage: whitelabel v0 release-notes [options] [command]

Options:
  -h, --help                                        display help for command

Commands:
  parse-release-notes [options]                     parse the release notes and outputs the release notes as JSON string and writes them to the specified file
  move-to [options] <new-version-name>              move the release notes in 'unreleased' to a new subdirectory <new-version-name>
  prepare-metadata [options] <appName> <storeName>  prepare metadata for store
  help [command]                                    display help for command
"
`;

exports[`stability test v0 1`] = `
"Usage: whitelabel v0 [options] [command]

Options:
  -h, --help      display help for command

Commands:
  release-notes
  version         Manage version
  build-config
  release
  ci
  notify
  help [command]  display help for command
"
`;

exports[`stability test version 1`] = `
"Usage: whitelabel v0 version [options] [command]

Manage version

Options:
  -h, --help      display help for command

Commands:
  calc            calculate the next version
  help [command]  display help for command
"
`;
